src/libstd/ffi/c_str.rs

```diff
@@ -274,7 +287,12 @@ impl CString {
     /// Failure to call `from_raw` will lead to a memory leak.
     #[stable(feature = "cstr_memory", since = "1.4.0")]
     pub fn into_raw(self) -> *mut c_char {
-        Box::into_raw(self.into_inner()) as *mut c_char
+        if let Some(bytes) = self.into_inner() {
+            if bytes.len() > 1 {
```

Comment:
I do not see a reason for this conditional. Return the pointer always. Alternatively assert that if bytes.len() == 1, the pointer is the same as of NUL_TERMINATED_EMPTY slice.